### PR TITLE
Gutenboarding: Check for a selected plan to determine whether to show the plans grid

### LINKED ISSUE #42218: Gutenboarding: do not show plans grid if a plan was manually selected
To reproduce:
1. Enter the flow (incognito, no account)
2. Click 'free plan' in the header
3. Pick a paid plan
4. After picking a design see the plans grid again

Expected:
Since I picked a plan I should advance straight to account creation, not see the plans grid again.

cc @johnHackworth

### PR DESCRIPTION
**Pull request by ramonjd**
## Changes proposed in this Pull Request

Check for a selected plan to determine whether to show the plans grid. 

This doesn't yet take into account whether:

- a user has already seen the plans page, in which case we could allow it to be shown

## Testing instructions

Enter the `/new` flow and carry on without selecting a plan. You should see the plans page.
Enter the `/new` flow and select a paid domain. You should see the plans page.
Enter the `/new` flow and select a plan from the modal. You should NOT see the plans page.

## TODO

- [x] Persist the plans page if the user has already seen and interacted with it (otherwise it might cause confusion?)
- [x] Entering the `/new/premium` does not select the plan in the grid. Should we do this?

Fixes #42218

### CODE REVIEW

**Comment by roo2** [comment]
I added the check so that if the user has seen the plans step once, they will continue to see it. This will make going back and forth in the flow work as the other pages instead of skipping the plan step if you've seen it once.

**Comment by roo2** [comment]
Also, when I access `/new/premium` it is already selecting the premium plan in the top right of the screen, the plans modal, and the plans step as I'd expect 👍

File: client/landing/gutenboarding/stores/onboard/reducer.ts [hunk 1]
```diff
@@ -137,11 +137,22 @@ const isRedirecting: Reducer< boolean, OnboardAction > = ( state = false, action
 	return state;
 };
 
+const hasUsedPlansStep: Reducer< boolean, OnboardAction > = ( state = false, action ) => {
```
Comment (razvanpapadopol) [inline client/landing/gutenboarding/stores/onboard/reducer.ts:140]: this new reducer has to be added to the persistence list in order to persist between 2 user visits: https://github.com/Automattic/wp-calypso/blob/b524c70eac79dfcbef7f5bbff1423fe48a232422/client/landing/gutenboarding/stores/onboard/index.ts#L27-L35

**Comment by simison** [comment]
Because we're changing mechanics of the flow, we need to change the test date here:

https://github.com/Automattic/wp-calypso/blob/0646c156ec0ff885eeab158127e9171a38e66aab/client/lib/abtest/active-tests.js#L119

@martinremy wrote elsewhere:

> if you all change the flow, can you also change the datestamp for the test.  That will give us a new test slug for the analysis, because it'll be essentially a new test.

File: client/landing/gutenboarding/stores/onboard/reducer.ts (hunk 1 above)
Comment (roo2) [reply client/landing/gutenboarding/stores/onboard/reducer.ts:140]: excellent, thank you!

**Comment by roo2** [comment]
> Because we're changing mechanics of the flow, we need to change the test date here:

Excellent, I updated it to the 15th `20200515`

**Pull request merged**